buildpipeline/perf-pipeline.groovy

```diff
@@ -6,9 +6,374 @@
+        
+        String baselineString = ""
+        if (isBaseline) {
+            baselineString = "-baseline"
+        }
+
+        String pgoTestFlag = ((pgo == 'nopgo') ? '-nopgo' : '')
+        
+        dir ('.') {
+            unstash "nt-${arch}-${pgo}${baselineString}-build-artifacts"
+            unstash "benchview-tools"
+            unstash "throughput-benchmarks-${arch}"
+            unstash "metadata"
+        }
+
+        // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+        // submission-metadata.py
+        if (isBaseline) {
+            bat "move /y submission-metadata-baseline.json submission-metadata.json"
+        }
+
+        bat "py \".\\Microsoft.BenchView.JSONFormat\\tools\\machinedata.py\""
+        bat ".\\init-tools.cmd"
+        bat "tests\\runtest.cmd ${config} ${arch} GenerateLayoutOnly"
+        bat "py -u tests\\scripts\\run-throughput-perf.py -arch ${arch} -os ${os} -configuration ${config} -opt_level ${optLevel} -jit_name ${jit} ${pgoTestFlag} -clr_root \"%WORKSPACE%\" -assembly_root \"%WORKSPACE%\\${arch}ThroughputBenchmarks\\lib\" -benchview_path \"%WORKSPACE%\\Microsoft.Benchview.JSONFormat\\tools\" -run_type ${runType}"
+        archiveArtifacts allowEmptyArchive: false, artifacts:'throughput-*.csv,machinedata.json'
+    }
+}
+
+//------------------------ Linux Functions ----------------------------//
+
+def linuxBuild(String arch, String config, String pgo, boolean isBaseline) {
+    checkout scm
+
+    String pgoBuildFlag = ((pgo == 'nopgo') ? '-nopgooptimize' : '')
+    String baselineString = ""
+
+    // For baseline runs, checkout the merge's parent
+    if (isBaseline) {
+        baselineString = "-baseline"
+        sh "git checkout HEAD^1"
+    }
+
+    sh "./build.sh -verbose -${config} -${arch} ${pgoBuildFlag}"
+    stash name: "linux-${arch}-${pgo}${baselineString}-build-artifacts", includes: 'bin/**'
+}
+
+def linuxPerf(String arch, String os, String config, String uploadString, String runType, String optLevel, String pgo, boolean isBaseline) {
+    withCredentials([string(credentialsId: 'CoreCLR Perf BenchView Sas', variable: 'BV_UPLOAD_SAS_TOKEN')]) {
+        checkout scm
+
+        String baselineString = ""
+        if (isBaseline) {
+            baselineString = "-baseline"
+        }
+
+        String pgoTestFlag = ((pgo == 'nopgo') ? '-nopgo' : '')
+
+        dir ('.') {
+            unstash "linux-${arch}-${pgo}${baselineString}-build-artifacts"
+            unstash "nt-${arch}-${pgo}${baselineString}-test-artifacts"
+            unstash "benchview-tools"
+            unstash "metadata"
+        }
+
+        // We want to use the baseline metadata for baseline runs. We expect to find the submission metadata in
+        // submission-metadata.py
+        if (isBaseline) {
+            sh "mv -f submission-metadata-baseline.json submission-metadata.json"
+        }
+
+        sh "./tests/scripts/perf-prep.sh"
```

Comment:
Since we are doing the work to pull down Benchview scripts and stashing them we should remove that stuff from perf-prep.sh.  It does a bunch of work to pull down tests that we don't need to do anymore.